main.rb

```diff
@@ -0,0 +1,179 @@
+module Enumerable
+  def my_each
+    i = 0
+    length.times do
+      yield (self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    length.times do
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    if block_given?
+      arr_ret = []
+      i = 0
+      length.times do
+        arr_ret.push(self[i]) if yield(self[i])
+        i += 1
+      end
+      arr_ret
+    end
+  end
+
```

Comment:
should have working `my_select` **(covered by automated tests)** identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))